packages/admin/src/pages/build-application/[applicationId]/[sectionId]/question-options.page.tsx

```diff
@@ -56,21 +56,32 @@ export const getServerSideProps: GetServerSideProps = async ({
     (section) => section.sectionId === sectionId
   )?.sectionTitle;
 
-  const result = await callServiceMethod(
-    req,
-    res,
-    async (body: any) => {
-      options = body.options;
+  function processBody(body: any, options: string[]) {
+    options = Object.keys(body).reduce((array, key) => {
+      if (key.startsWith('options')) {
+        array.push(...body[key]);
+      }
 
-      if ('add-another-option' in body) {
+      if (key.startsWith('delete_')) {
+        const deleteOptionIndex = Number(key.split('_')[1]);
+        array.splice(deleteOptionIndex, 1);
+      }
+      return array;
+    }, [] as string[]);
+    return options;
+  }
+
+  async function handleOptions(body: any, options: string[]) {
+    const { optional, ...restOfQuestionSummary } = questionSummary;
+    // switch expression 'true' for evaluation of boolean case labels
+    switch (true) {
```

Comment:
i think this should be an if-else, it reduces the amount of indentation with no cost to readability